src/d/actor/d_a_obj_rflw.cpp

```diff
@@ -40,50 +42,166 @@ static dCcD_SrcCyl l_cyl_src = {
+    JUT_ASSERT(0xAA, modelData != NULL);
+
+    mpModel = mDoExt_J3DModel__create(modelData, 0, 0x11020203);
+
+    if (mpModel != NULL) {
+        JUTNameTab* name_table = mpModel->getModelData()->getJointName();
+        for (u16 i = 0; i < mpModel->getModelData()->getJointNum(); i++) {
+            if (strcmp("joint2", name_table->getName(i)) == 0) {
+                mpModel->getModelData()->getJointNodePointer(i)->setCallBack(nodeCallBack);
+                break;
+            }
+        }
+        mpModel->setUserArea(reinterpret_cast<u32>(this));
+    } else {
+        return FALSE;
+    }
+
+    return TRUE;
 }
 
 /* 000001E0-00000284       .text CreateInit__11daObjRflw_cFv */
 void daObjRflw_c::CreateInit() {
-    /* Nonmatching */
+    fopAcM_SetMtx(this, mpModel->getBaseTRMtx());
+    fopAcM_setCullSizeBox(this, -600.0f, -0.0f, -600.0f, 600.0f, 900.0f, 600.0f);
+    fopAcM_setCullSizeFar(this, 1.0f);
+    mStts.Init(0xFF, 0xFF, this);
+    mCyl.Set(l_cyl_src);
+    mCyl.SetStts(&mStts);
+    m408 = 0;
+    set_mtx();
 }
 
 /* 00000284-00000368       .text nodeCallBack__FP7J3DNodei */
-static BOOL nodeCallBack(J3DNode*, int) {
-    /* Nonmatching */
+static BOOL nodeCallBack(J3DNode* node, int calcTiming) {
+    if (calcTiming == J3DNodeCBCalcTiming_In) {
+        J3DJoint* joint = (J3DJoint*)node;
+        s32 jointNo = joint->getJntNo();
+        J3DModel* model = j3dSys.getModel();
+        daObjRflw_c* i_this = (daObjRflw_c*)model->getUserArea();
+        if (i_this != NULL) {
+            MTXCopy(model->getAnmMtx(jointNo), *calc_mtx);
+            cMtx_XrotM(*calc_mtx, i_this->m40E);
+            cMtx_YrotM(*calc_mtx, i_this->m408);
+            cMtx_XrotM(*calc_mtx, -i_this->m40E);
+            model->setAnmMtx(jointNo, *calc_mtx);
+            MTXCopy(*calc_mtx, J3DSys::mCurrentMtx);
+        }
+    }
+    return TRUE;
 }
 
 /* 00000368-000003E8       .text set_mtx__11daObjRflw_cFv */
 void daObjRflw_c::set_mtx() {
-    /* Nonmatching */
+    mpModel->setBaseScale(scale);
+    mDoMtx_stack_c::transS(current.pos);
+    mDoMtx_stack_c::YrotM(current.angle.y);
+    mpModel->setBaseTRMtx(mDoMtx_stack_c::get());
+}
+
+inline cPhs_State daObjRflw_c::_create() {
+    cPhs_State rt = cPhs_ERROR_e;
+    fopAcM_SetupActor(this, daObjRflw_c);
+
+    rt = dComIfG_resLoad(&mPhs, "Rflw");
+
+    if (rt == cPhs_COMPLEATE_e) {
+        if (!fopAcM_entrySolidHeap(this, CheckCreateHeap, 0xC60)) {
+            return cPhs_ERROR_e;
+        }
+        CreateInit();
+    }
+    m410 = false;
+    return rt;
+}
+
+inline BOOL daObjRflw_c::_delete() {
+#if VERSION == VERSION_DEMO
+    dComIfG_deleteObjectRes("Rflw");
+#else
+    dComIfG_resDelete(&mPhs, "Rflw");
+#endif
+    return TRUE;
+}
+
+inline BOOL daObjRflw_c::_draw() {
+    dKy_tevstr_c* p_tev_str;
+    g_env_light.settingTevStruct(TEV_TYPE_BG0, &current.pos, p_tev_str = &tevStr); // needed to do this to match
```

Comment:
Just to make it easier to ctrl+f for these later
```suggestion
    g_env_light.settingTevStruct(TEV_TYPE_BG0, &current.pos, p_tev_str = &tevStr); // fakematch
```